### PR TITLE
Rename /builder/* paths to /tekton/*

### PR DESCRIPTION
**Pull request by ImJasonH**
Related to #1030 

This changes the paths Tekton reserves under  `/builder/*` for various reasons, to paths under `/tekton/*`.

This doesn't (yet) document what paths those are and what they're for, it's just a simple search-and-replace:

```
git grep -l \/builder\/ | xargs sed -i -e 's/\/builder\//\/tekton\//g'
```

This change might constitute a breaking change, since some users (including, notably, some of our release tasks) might have logic that depends on the exact path string. When we better document reserved paths, we should also document how to avoid relying on their specifics.

This change also simplifies `publish-nightly.yaml` which persisted some data into `/builder/home` just so that it would be present in the next step -- instead, I just combined the steps into one. I can remove this change or move it to a separate PR if anybody prefers.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [y] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [y] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [y] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Renamed Tekton-reserved paths under /builder/* to be under /tekton/* instead. This might break some users who depended on the specific path being under /builder/*
```

### CODE REVIEW

**Comment by ImJasonH** [comment]
/hold 

Waiting until after the 0.8 release cut

**Comment by afrittoli** [comment]
@ImJasonH this may need a rebase 👼 
Any chance we could do this as a two step change, non breaking first, i.e. first add /tekton and link /builder to /tekton, and then in v0.10.0 drop the link? If this can be done, I'm happy to wait for it until Friday so that it can make v0.9.0.

I think it would be nice to do this soon, so that it has time to settle down before we reach beta.

@vdemeester @abayer was there any feedback about this possibly breaking j-x?

File: examples/taskruns/task-multiple-output-image.yaml [hunk 1]
```diff
@@ -73,8 +73,8 @@ spec:
     - -ce
     - |
       set -ex
-      mkdir -p /builder/home/image-outputs/builtImage1
-      cat <<EOF > /builder/home/image-outputs/builtImage1/index.json
+      mkdir -p /tekton/home/image-outputs/builtImage1
```
Comment (afrittoli) [inline examples/taskruns/task-multiple-output-image.yaml]: NIT: I think a better change here would be to not use a fixed path at all, instead use a variable. Something for another patch probably.

File: examples/taskruns/task-multiple-output-image.yaml [hunk 2]
```diff
@@ -94,8 +94,8 @@ spec:
     - -ce
     - |
       set -e
-      mkdir -p /builder/home/image-outputs/builtImage2
-      cat <<EOF > /builder/home/image-outputs/builtImage2/index.json
+      mkdir -p /tekton/home/image-outputs/builtImage2
+      cat <<EOF > /tekton/home/image-outputs/builtImage2/index.json
```
Comment (afrittoli) [inline examples/taskruns/task-multiple-output-image.yaml]: Ditto

**Comment by ImJasonH** [comment]
> @ImJasonH this may need a rebase 👼
> Any chance we could do this as a two step change, non breaking first, i.e. first add /tekton and link /builder to /tekton, and then in v0.10.0 drop the link? If this can be done, I'm happy to wait for it until Friday so that it can make v0.9.0.
> 
> I think it would be nice to do this soon, so that it has time to settle down before we reach beta.

Yeah, I was thinking about splitting it into two changes actually:

- `/builder/tools`, `/builder/scripts` -> `/tekton/` immediately, since those are internal implementation details that nobody should be relying on today.
- `/builder/home` -> `/tekton/` _and_ `/builder/` for a release (with `HOME` pointing to `/tekton/home`), then removed from `/builder/` in the next release.

How's that sound?

I'll try to get to that today or at least this week, before the release cutoff. The first one should be easy and should definitely make 0.9.

**Comment by vdemeester** [comment]
> Yeah, I was thinking about splitting it into two changes actually:
> 
>     * `/builder/tools`, `/builder/scripts` -> `/tekton/` immediately, since those are internal implementation details that nobody should be relying on today.
> 
>     * `/builder/home` -> `/tekton/` _and_ `/builder/` for a release (with `HOME` pointing to `/tekton/home`), then removed from `/builder/` in the next release.
> 
> 
> How's that sound?

I like that idea a lot :angel: The first one would be a requirement for 0.9 :wink:

**Comment by ImJasonH** [comment]
Part 1 in #1626

**Pull request closed without merging**